drivers/firmware/tisci/tisci.c

```diff
@@ -0,0 +1,1600 @@
+/*
+ * Copyright (c) 2025, Texas Instruments
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#define DT_DRV_COMPAT ti_k2g_sci
+#include <zephyr/drivers/mbox.h>
+#include <zephyr/device.h>
+#include "tisci.h"
+#include <zephyr/drivers/firmware/tisci/tisci.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#define LOG_LEVEL CONFIG_MBOX_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ti_k2g_sci);
+
+/**
+ * @struct tisci_config - TISCI device configuration
+ * @param mbox_tx: Mailbox transmit channel
+ * @param mbox_rx: Mailbox receive channel
+ * @param host_id: Host ID for the device
+ */
+struct tisci_config {
+	struct mbox_dt_spec mbox_tx;
+	struct mbox_dt_spec mbox_rx;
+	uint32_t host_id;
+};
+
+static struct k_sem response_ready_sem;
+
+/* Received message container */
+static struct rx_msg rx_message;
+static uint8_t rx_message_buf[MAILBOX_MBOX_SIZE];
+
+/**
+ * @struct tisci_xfer - TISCI transfer details
+ * @param tx_message: Transmit message
+ * @param rx_message: Received message
+ */
+struct tisci_xfer {
+	struct mbox_msg tx_message;
+	struct rx_msg rx_message;
+};
+
+/**
+ * @struct tisci_data - Runtime data for TISCI device
+ * @param xfer: Current transfer details
+ * @param desc: Device descriptor
+ * @param version: Firmware version info
+ * @param host_id: Host ID for the device
+ * @param seq: Current transfer sequence number
+ */
+struct tisci_data {
+	struct tisci_xfer xfer;
+	struct tisci_desc desc;
+	struct tisci_version_info version;
+	uint32_t host_id;
```

Comment:
`host_id` is static for each instance, should go in config. Actually it is already in config also, so no need for this second copy.